### PR TITLE
Throw custom errors from relayer

### PR DESCRIPTION
**Pull request by iamacook**
This adds custom `Error`s with respective `ExceptionFilter`s for the following to the relayer:

- Invalid `multiSend`
- Invalid transfers
- Unofficial mastercopies
- Unofficial MultiSend contracts

### CODE REVIEW

File: src/domain/relay/errors/invalid-multisend.error.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+export class InvalidMultiSendError extends Error {
+  constructor() {
+    super(
+      'Invalid `multiSend` call. The batch is not all `execTransaction` calls to same address.',
```
Comment (fmrsabino) [inline src/domain/relay/errors/invalid-multisend.error.ts:4]: nitpick: I'd remove the ` as I assume this is mostly for rendering in markdown.

File: src/domain/relay/exception-filters/invalid-transfer.exception-filter.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { InvalidTransferError } from '@/domain/relay/errors/invalid-transfer.error';
+
+@Catch(InvalidTransferError)
+export class InvalidTransferExceptionFilter implements ExceptionFilter {
+  catch(_: InvalidTransferError, host: ArgumentsHost): void {
+    const ctx = host.switchToHttp();
+    const response = ctx.getResponse<Response>();
+
+    response.status(HttpStatus.UNPROCESSABLE_ENTITY).json({
+      message: 'Invalid transfer.',
```
Comment (fmrsabino) [inline src/domain/relay/exception-filters/invalid-transfer.exception-filter.ts:17]: We can use the error message here as it seems to provide even more context. Same as with the others.

File: src/domain/relay/errors/unofficial-mastercopy.error.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export class UnofficialMastercopyError extends Error {
```
Comment (fmrsabino) [inline src/domain/relay/errors/unofficial-mastercopy.error.ts:1]: nitpick: `UnofficialMasterCopyError`

File: src/domain/relay/errors/invalid-multisend.error.ts (hunk 1 above)
Comment (iamacook) [reply src/domain/relay/errors/invalid-multisend.error.ts]: Removed in 4a9d3787f8f6f122c9eb31a6e5e4cb00acc1c6e4.

File: src/domain/relay/exception-filters/invalid-transfer.exception-filter.ts (hunk 2 above)
Comment (iamacook) [reply src/domain/relay/exception-filters/invalid-transfer.exception-filter.ts]: Changed in 4a9d3787f8f6f122c9eb31a6e5e4cb00acc1c6e4.

File: src/domain/relay/errors/unofficial-mastercopy.error.ts (hunk 3 above)
Comment (iamacook) [reply src/domain/relay/errors/unofficial-mastercopy.error.ts]: Changed in 4a9d3787f8f6f122c9eb31a6e5e4cb00acc1c6e4.

File: src/datasources/cache/cache.router.ts [hunk 4]
```diff
@@ -407,7 +408,8 @@ export class CacheRouter {
   }
 
   static getRelayKey(args: { chainId: string; address: string }): string {
-    return `${args.chainId}_${CacheRouter.RELAY_KEY}_${args.address}`;
+    // Ensure address is checksummed to always have a consistent cache key
+    return `${args.chainId}_${CacheRouter.RELAY_KEY}_${getAddress(args.address)}`;
```
Comment (fmrsabino) [inline src/datasources/cache/cache.router.ts:412]: Missed this one: I think this should be outside of the scope of this PR. Also, at this level we should be as close as possible to Redis primitives. Setting the key as a checksummed address should happen in outer layers (that handle more ethereum related logic).

**Review by fmrsabino — CHANGES_REQUESTED** [review]

File: src/datasources/cache/cache.router.ts (hunk 4 above)
Comment (iamacook) [reply src/datasources/cache/cache.router.ts]: Agreed. This is a remnant from an older change that shouldn't be there. I manually removed it in 7c9718a097a267c29733988e72fb26f7f5971361.

**Review by fmrsabino — APPROVED** [review]
🚀

**Pull request merged**